File: JavaToKotlin/app/src/main/java/io/dustin/apps/board/domain/bookmark/model/Bookmark.java
package io.dustin.apps.board.domain.bookmark.model;

import io.dustin.apps.common.model.BaseEntity;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

@Getter
@Entity
@DynamicUpdate
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Bookmark extends BaseEntity {

    @Builder
    public Bookmark(Long id, @NotNull Long userId, @NotNull Long boardId){
        this.id =id;
        this.userId = userId;
        this.boardId = boardId;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id")
    private Long userId;

    @Column(name = "board_id")
    private Long boardId;
}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/bookmark/model/Bookmark.kt
package io.dustin.apps.board.domain.bookmark.model

import io.dustin.apps.common.model.BaseEntity
import jakarta.persistence.*
import org.hibernate.annotations.DynamicUpdate

@Entity
class Bookmark (

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    val userId: Long,

    val boardId: Long,

) : BaseEntity()